### PR TITLE
Complex rework

### PR DESCRIPTION
**Pull request by jamescassell**
followup to #145
also fixes some fatal errors we ran into with assert tasks that should have been debug tasks

### CODE REVIEW

File: tasks/fix-cat2.yml [hunk 1]
```diff
@@ -1015,12 +1023,16 @@
       - name: "MEDIUM | RHEL-07-020900 | AUDIT | All system device files must be correctly labeled to prevent unauthorized modification."
         debug:
             msg: "{{ rhel_07_020900_audit.stdout_lines }}"
+        changed_when:
+            - rhel7stig_audit_complex
```
Comment (shepdelacreme) [inline tasks/fix-cat2.yml]: Should there be a conditional here to only show changed when `rhel_07_020900_audit.stdout_lines` contains something?

File: tasks/fix-cat2.yml [hunk 2]
```diff
@@ -1603,10 +1615,15 @@
       - name: "MEDIUM | RHEL-07-032010 | AUDIT | The system must update the DoD-approved virus scan program every seven days or more frequently."
         debug:
             msg: "{{ rhel_07_032010_audit.stdout_lines }}"
+        changed_when:
+            - rhel7stig_audit_complex
```
Comment (shepdelacreme) [inline tasks/fix-cat2.yml]: Should there be a conditional here to only show changed when rhel_07_032010_audit.stdout_lines contains something?

File: tasks/fix-cat2.yml (hunk 1 above)
Comment (jamescassell) [reply tasks/fix-cat2.yml]: agreed

File: tasks/fix-cat2.yml (hunk 2 above)
Comment (jamescassell) [reply tasks/fix-cat2.yml]: likewise

**Comment by jamescassell** [comment]
@shepdelacreme good catch on the idempotence.  I've pushed one more commit that should fix that.

**Comment by shepdelacreme** [comment]
LGTM! Thanks!

**Pull request merged**